src/index.tsx

```diff
@@ -16,50 +16,97 @@ import {
+        this.history.replace({ ...this.history.location, search: stringify(params) });
+    }
+
+    onParamRemoved(key: string, callback: () => void) {
+        let hasParam = false;
+        this.history.listen((location) => {
+            const params = parse(location.search, { ignoreQueryPrefix: true });
+            if (params[key]) {
+                hasParam = true;
+            } else if (hasParam) {
+                callback();
+                hasParam = false;
+            }
+        })
+    }
+}
+
+const params = new URLParams(createBrowserHistory());
 const initialSelections = {
     ...initialState,
-    ...UrlState.toAppState(pram.getParams())
+    ...UrlState.toAppState(params.getParams())
 }
 const store = createReduxStore({ selection: initialSelections });
 
-store.dispatch(selection.actions.syncStateWithURL(pram.getParams()));
+store.dispatch(selection.actions.syncStateWithURL(params.getParams()));
 
 // when the dataset query param changes, check if it's been removed
-pram.onChange("dataset", function notifyHostFrameOfURLChange(dataset) {
-    if (!dataset) {
-        // used the back button to get back to the landing page
-        // UrlState doesn't sync actions if a query param doesn't exist, 
-        // so clearing it out here 
-        // NOTE: order matters, the selections need to be cleared out before the 
-        // dataset metadata is cleared
-        store.dispatch(selection.actions.clearDataset());
-        store.dispatch(metadata.actions.clearDatasetValues());
-    }
+params.onParamRemoved("dataset", () => {
+    // used the back button to get back to the landing page
+    // UrlState doesn't sync actions if a query param doesn't exist, 
+    // so clearing it out here 
+    // NOTE: order matters, the selections need to be cleared out before the 
+    // dataset metadata is cleared
+    store.dispatch(selection.actions.clearDataset());
+    store.dispatch(metadata.actions.clearDatasetValues());
 });
 
 // keep the URL in sync with the state of this app
 store.subscribe(function updateURL() {
     const state = store.getState();
     const mapping = UrlState.toUrlSearchParameterMap(state.selection);
-    const currentParams = pram.getParams();
+    const currentParams = params.getParams();
     const datasetChanged = UrlState.paramChanged(URLSearchParam.dataset, mapping, currentParams);
     if (datasetChanged) {
         // only save in browser history if the dataset is different
-        pram.pushParam(URLSearchParam.dataset, mapping.dataset);
+        params.pushParam(URLSearchParam.dataset, mapping.dataset);
     }
     
     if (UrlState.paramsChanged(mapping, currentParams)) {
-        pram.replaceParams(mapping);
+        params.replaceParams(mapping);
     }
 });
 
-render(
-    <Provider store={store}>
-        <App />
-    </Provider>,
-    document.getElementById(APP_ID)
-);
+const container = document.getElementById(APP_ID);
+if (container === null) {
+    console.error("App container missing!");
```

Comment:
maybe just print the APP_ID value too, if this kind of error really can happen.

Reply:
Seems unlikely (this is just motivated by a type error) but worth doing regardless.